crates/evm/src/execute.rs

```diff
@@ -499,6 +502,46 @@ where
     }
 }
 
+/// Creates an `EvmState` from a map of balance increments and the current state
+/// to load accounts from.
+/// Zero balance increments are ignored and won't create state entries.
+pub fn balance_increment_state<DB>(
+    balance_increments: &HashMap<Address, u128, DefaultHashBuilder>,
+    state: &mut State<DB>,
+) -> Result<EvmState, BlockExecutionError>
+where
+    DB: Database,
+{
+    let mut load_account = |address: &Address| -> Result<(Address, Account), BlockExecutionError> {
+        let cache_account = state.load_cache_account(*address).map_err(|_| {
+            BlockExecutionError::msg("could not load account for balance increment")
+        })?;
+
+        let account = cache_account.account.as_ref().ok_or_else(|| {
+            BlockExecutionError::msg("could not load account for balance increment")
+        })?;
+
+        let storage = account
+            .storage
+            .iter()
+            .map(|(k, v)| {
+                (*k, EvmStorageSlot { original_value: *v, present_value: *v, is_cold: false })
+            })
+            .collect();
```

Comment:
as i've mentioned, this is unnecessary if we leave the fn as is

Reply:
sure done here https://github.com/paradigmxyz/reth/pull/13050/commits/7cf98ee89ae5ba15f277607c8422949e07434541